### PR TITLE
[#1703] Upgrade Node.js to 14

### PR DESCRIPTION
**Pull request by yhtMinceraft1010X**
Part of #1703

## Proposed commit message

```
The node versions used in GitHub Actions and build.gradle do not match.
At the same time, Node.js 10 has reached end of life, while Node.js 12
will reach end of life on 30 April 2022.

Let's update Node.js to use Node 14.
```

## Other information

The long-term goal is to upgrade to Node 16. However, that requires sorting out and updating the dependencies in the `package.json`.

### CODE REVIEW

**Comment by gok99** [comment]
LGTM! Checks for this PR are using the new workflow and are all passing.

**Review by gok99 — APPROVED** [review]

**Comment by gerhean** [comment]
The funny thing is that the workflow doesn't actually need to install node as `com.liferay.node` installs another node. The only thing that requires external node is the hotreload functionality, which probably should be looked into.

File: .github/workflows/gh-pages.yml [hunk 1]
```diff
@@ -19,9 +19,9 @@ jobs:
       uses: actions/checkout@v2
 
     - name: Set up Node.js
-      uses: actions/setup-node@v1
+      uses: actions/setup-node@v2
       with:
-        node-version: '12.x'
+        node-version: '14.19.0'
```
Comment (gerhean) [inline .github/workflows/gh-pages.yml:24]: Would it be better to use `14.x` instead?

[the remark this answers is not in the document]
Comment (gerhean) [reply .github/workflows/gh-pages.yml:24]: The long-term goal is to upgrade to Node 16. However, that requires sorting out and updating the dependencies in the `package.json`

**Comment by gerhean** [comment]
Actually instead of slowly upgrading, why not upgrade everything at once?

**Comment by yhtMinceraft1010X** [comment]
One of the current roadblocks to upgrading straight to Node 16 is that some dependencies give errors. Iirc, one of them is `eslint-config-airbnb-base` version `13.2.0` and `eslint` version `6.8.0`. All the dependencies need to be upgraded and some involve changes to frontend linting and the vue files.

Hence why I'm opening a PR for Node 14 to "close the gap" while trying to figure out the Node 16 dependencies.

**Review by dcshzj — APPROVED** [review]
LGTM!

**Pull request merged**